File: aizu/ALDS1/BubbleSort.cpp
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <string.h>

void echo(int array[], int num) {
    for(int i = 0; i < num; i++) {
        if(i > 0) printf(" ");
        printf("%d", array[i]);
    }
    printf("\n");
}

int sort(int array[], int num) {
    int tmp = 0;
    int cnt = 0;
    int flg = 1;
    int j;

    
    while (flg == 1) {
        flg = 0;
        for (int i = num - 1; i >= 1; i--) {
            if (array[i] < array[i - 1]) {
                tmp = array[i];
                array[i] = array[i - 1];
                array[i - 1] = tmp;
                flg = 1;
                cnt++;
            }
        }
    }

    return cnt;
}

int main(void) {
    int num = 0;
    int cnt = 0;
    
    scanf("%d", &num);

    int *array = (int *)malloc(num * sizeof(int));

    for (int i = 0; i < num; i++) {
        scanf("%d", &array[i]);
    }

    cnt = sort(array, num);
    
    echo(array, num);
    printf("%d\n", cnt);
    
    free(array);

    return 0;
}

File: aizu/ALDS1/ALDS1_2_A/BubbleSort.py
class BubbleSort:
    def __init__(self, A = [], N = 0):
        self.A = A
        self.N = N
        pass

    def sort(self, A = [], N = 0):
        if A != None:
            self.A = A
        if N != 0:
            self.N = N
        if A == []:
            return A
        flg = True
        c = 0
        while flg:
            flg = False
            for j in range(1, self.N):
                if self.A[j] < self.A[j - 1]:
                    self.A[j], self.A[j - 1] = self.A[j - 1], self.A[j]
                    flg = True
                    c += 1
        for j in range(self.N):
            if j == self.N - 1:
                print(self.A[j])
            else:
                print(self.A[j], end=' ')
        print(c)
        return c, A